# Learning Outcome 3: Assess the general quality of the raw data from the sequencing facility

## LO 3.1 - Interpret what are fastq files and what is their content

Most high-throughput sequencing (HTS) machines output [fastq files](https://en.wikipedia.org/wiki/FASTQ_format), the “de facto” current standard in HTS. Fastq files are simply text files, where each block of information (a sequenced DNA fragment, or read) in this format is encoded as 4 lines:

	@read_identifier
	read_sequence
	+ separator line
	base_qualities

Each base has a quality character associated with it, which represents how confidently the
machine identified (called) the base. The probability of error per base is given as a [Phred score](https://en.wikipedia.org/wiki/Phred_quality_score), calculated from an integer value (Q) derived from the quality character associated to the base. Useful reference values of Q include:
* Q=10 - 90% accuracy
* Q=20 - 99% accuracy
* Q=30 - 99.9% accuracy
* Q=40 - 99.99% accuracy

Although there's theoretically no limit, Q usually goes up to around 40 in recent illumina machines.

You can see a few fastq files in the folder fastq_examples:
* sample_quality_and_adaptors.fastq.gz
* sample_adaptors.fastq.gz
* 20150821.A-2_BGVR_P218_R1.sample.fastq.gz
* 20150821.A-2_BGVR_P218_R2.sample.fastq.gz

Since each fastq can have several million reads, they can become very big. Therefore, it is usual to keep them in a compressed format such as gzip. Most recent software can directly read compressed fastq files.

**Task**: Upload all the sample files into your Galaxy and inspect them

You probably noticed that two of the example files have the same name, except for R1 and R2. This is an example of a paired-end dataset. If you inspect both datasets, you can find the same identifiers in each of the files, in the same order. In R1 you have forward reading of a fragment, and in R2 you have the reverse reading of the same fragment. Depending on the length of the fragment, they may or may not contain overlapping sequences. 


## LO 3.2 - Use software like FastQC to process fastq files and produce QC reports

High Throughput Sequencing machines read thousands or millions of sequences in paralell. As you can imagine, this usually generates large fastq files, with millions of lines, and thus manually inspecting the quality of each read is out of the question. Specialized software has been developed to provide quality measures for fastq files generated by HTS machines. [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) is a popular program to generate quality reports on fastq data. Running FastQC on your raw data is usually the first thing you should do once you receive a new dataset.

**Task**: Run FastQC in each of the example files


## LO 3.3 - Read QC reports of raw data to assess the general quality of data and presence of sequence bias

**Task**: Inspect the FastQC Reports generated previously

Detect low quality bases in the QC reports

Detect sequence bias and possible presence of adaptors and other contaminants

The FastQC reports provide a series of plots that allow the user to assess the overall quality of their raw data and detect potential biases and problems. 

One of the plots indicates the distribution of base qualities (as Q values representing the confidence in the base) along the length of the reads. You can notice that, at least for illumina data, on average the quality of each base tends to decrease along the length of the read. The beginning of the read may also be problematic. You can also see that the reverse read (R2) is usually of worse quality than the forward read (R1). Therefore, short single-end reads usually have better average quality, and are often ready to use right out of the sequencer.

[images of quality]



Probably the most relevant are .


When comparing fastq files generated from different technologies, you can see differences in their
properties. Illumina machines generate shorter reads, usually all with the same length (before filtering).
Pacbio and nanopore generate (much) longer reads, with diverse read lengths, and of a poorer quality.
Illumina generates many more reads, making both technologies complementary to each other (this will
become clearer when we look at specific applications). Finally, you can also notice that, independently
of the technology, the quality of base quality tends to decrease along the length of the read



# Learning Outcome 4: Do simple processing operations in the raw data to improve its quality

## LO 4.1 - Use tools such as seqtk and trimmomatic to remove low quality bases from your reads



## LO 4.2 - Use tools such as cutadapt to remove adaptors and other artefactual sequences from your reads


This is rev comp of adaptor: AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC


Overall, you can use Trimmommatic to do both quality and adaptor trimming. 



# Learning Outcome 5: Generate alignments of processed reads against a reference genome

## LO 5.1 - What is a reference genome, versioning and where to obtain genomes

			Question: Are genomes constant?
			Obtain genome fasta from Ensembl

## LO 5.2 - Alignment software: tophat2/hisat2; bwa; sailfish/salmon

			Question: what are the conditions of using burrows-wheeler approaches?	
			Prepare a reference genome to use with hisat2 and bwa

## LO 5.3 - Run an alignment: the SAM/BAM alignment format

			Run hisat2 / bwa mem in an example dataset
			Question: what is the SAM format; what is the BAM format?
	
# Learning Outcome 6: Assess the general quality of the alignments and detect possible problems

		LO 6.1 - Visualizing alignments in IGV for single genes

		LO 6.2 - Use tools such as RSeQC and Qualimap to assess quality of alignments
			Interpret general alignment statistics such as percentage of aligned reads
			Check the reports to assess RNA integrity and diversity

[Show example plots of some IGC datasets, without showing the actual data!]


# Learning Outcome 7: Generate tables of counts using the alignment and a reference gene annotation

## LO 7.1 - What is a reference gene annotation, versioning and where to obtain
	
		Question: what is the GFF/GTF format?
		Obtain genome GTF from Ensembl

## LO 7.2 - The process of generating gene counts from genome aligments

			Question: what parameters we need to consider when counting?

## LO 7.3 - Use tools such as htseq-counts to generate table of gene counts


# Learning Outcome 8: Generate lists of differentially expressed genes, at least for a simple pairwise comparison

## LO 8.1 - Using the R package edgeR to produce a pairwise differential expression analysis
	
		Use Galaxy to produce differentially expressed genes with edgeR
		Use edgeR in R and RStudio 

## LO 8.2 - Interpretation and visualization of results

		Produce PCA plots comparing all samples: outlier detection
		Visualize expression profiles of top differentially expressed genes
		Produce other plots such as vulcano plots

## LO 8.3 - Use more complex settings: Generalized Linear Models

		Account for confounders using Generalized Linear Models
		Performing ANOVA-like comparisons


# Learning Outcome 9 - Perform simple functional enrichment analysis and understand the concepts behind them

		(TODO: Daniel Faria)
		LO 9.1 - Functional annotations: what are these and where to get them
		LO 9.2 - The statistics behind functional enrichment analysis
		LO 9.3 - Using functional enrichment analysis with your lists of genes






