---
title: "Course Program"
author: "Daniel Sobral"
---


# Mon, April 17th

+ 09:30 - 10:00 Introduction to the course and self presentation of the participants
+ 10:00 - 11:00 Possibilities and limitations of NGS sequencing technologies. Choose adequate sequencing for your biological question
+ 11:00 - 11:30 ***Coffee Break***
+ 11:30 - 12:30 Steps in the analysis of RNA-Seq differential expression experiments
+ 12:30 - 14:00 ***LUNCH BREAK***
+ 14:00 - 16:00 Interpret what are fastq files and what is their content. Use software like FastQC to process fastq files and produce QC reports. Read QC reports of raw data to assess the general quality of data and presence of sequence bias.
+ 16:00 - 16:30 ***Tea Break***
+ 16:30 - 18:00 Use tools such as seqtk and trimmomatic to remove low quality bases from your reads. Use tools such as cutadapt to remove adaptors and other artefactual sequences from your reads 

# Tue, April 18th

+ 09:30 - 10:00 Morning Wrap-up (what have we done so far?)
+ 10:00 - 11:00 What is a reference genome, versioning and where to obtain genomes. Alignment software: tophat2/hisat2; bwa; sailfish/salmon. Run an alignment: the SAM/BAM alignment format. 
+ 11:00 - 11:30 ***Coffee Break***
+ 11:30 - 12:30 Visualizing alignments in IGV for single genes. Use tools such as RSeQC to assess quality of alignments
+ 12:30 - 14:00 ***LUNCH BREAK***
+ 14:00 - 16:00 What is a reference gene annotation, versioning and where to obtain. The process of generating gene counts from genome aligments. Use tools such as htseq-counts to generate table of gene counts.
+ 16:00 - 16:30 ***Tea Break***
+ 16:30 - 18:00 Using the R package edgeR to produce a pairwise differential expression analysis


# Wed, April 19th

+ 09:30 - 10:00 Morning Wrap-up (what have we done so far?)
+ 10:00 - 11:00 Use edgeR in R and RStudio. 
+ 11:00 - 11:30 ***Coffee Break***
+ 11:30 - 12:30 Interpretation and visualization of results.
+ 12:30 - 14:00 ***LUNCH BREAK***
+ 14:00 - 16:00 Interpretation and visualization of results.
+ 16:00 - 16:30 ***Tea Break***
+ 16:30 - 18:00 Use more complex settings: Generalized Linear Models


# Thu, April 20th

+ 09:30 - 10:00 Morning wrap-up (what have we done so far?)
+ 10:00 - 11:00 Use more complex settings: Generalized Linear Models
+ 11:00 - 11:30 ***Coffee Break***
+ 11:30 - 12:30 Functional annotations: what are these and where to get them. The statistics behind functional enrichment analysis.
+ 12:30 - 14:00 ***LUNCH BREAK***
+ 14:00 - 16:00 Using functional enrichment analysis with your lists of genes.
+ 16:00 - 16:30 ***Tea Break***
+ 16:30 - 18:00 Final wrap-up Session.
