---
title: "Course Program"
author: "Daniel Sobral"
---


# Mon, April 17th

+ 09:30 - 10:00 Introduction to the course and self presentation of the participants
+ 10:00 - 11:00 Possibilities and limitations of NGS sequencing technologies.
Choose adequate sequencing for your biological question
+ 11:00 - 11:30 ***Coffee Break***
+ 11:30 - 12:30 Steps in the analysis of RNA-Seq differential expression experiments
+ 12:30 - 14:00 ***LUNCH BREAK***
+ 14:00 - 16:00 Interpret what are fastq files and what is their content.
Use software like FastQC to process fastq files and produce QC reports.
Read QC reports of raw data to assess the general quality of data and presence of sequence bias.
+ 16:00 - 16:30 ***Tea Break***
+ 16:30 - 18:00 Improving the quality of raw sequencing data 

# Tue, April 18th

+ 09:30 - 10:00 Morning Wrap-up (what have we done so far?)
+ 10:00 - 11:00 Read Alignment and Assessment of Data Quality
+ 11:00 - 11:30 ***Coffee Break***
+ 11:30 - 12:30 From Aligned Reads to Gene Counts
+ 12:30 - 14:00 ***LUNCH BREAK***
+ 14:00 - 16:00 Linear Models and Experimental Design
+ 16:00 - 16:30 ***Tea Break***
+ 16:30 - 18:00 Introduction to RNA-seq and going from 'reads' to 'counts'


# Wed, April 19th

+ 09:30 - 10:00 Morning Wrap-up (what have we done so far?)

+ 09:30 - 11:00 Morning wrap-up (what have we done so far?)
                Differential Expression for RNA-seq Data
+ 11:00 - 11:30 ***Coffee Break***
+ 11:30 - 12:30 Continued RNA-seq practical
+ 12:30 - 14:00 ***LUNCH BREAK***
+ 14:00 - 16:00 Genome Annotation resources in Bioconductor
+ 16:00 - 16:30 ***Tea Break***
+ 16:30 - 18:00 Downstream analysis of RNA-seq analysis


# Thu, April 20th

+ 09:30 - 11:00 Morning wrap-up (what have we done so far?)
                Introduction to ChIP-seq
+ 11:00 - 11:30 ***Coffee Break***
+ 11:30 - 12:30 Practical: Quality assessment of ChIP-seq
+ 12:30 - 14:00 ***LUNCH BREAK***
+ 14:00 - 16:00 Downstream analysis of ChIP-seq
+ 16:00 - 16:30 ***Tea Break***
+ 16:30 - 18:00 Final wrap-up Session
